pint_models/database.py

```diff
@@ -0,0 +1,210 @@
+
+    Args:
+        dbconfig: (dict): A dictionary of config settings
+            that are required to connect to the Postgres DB
+
+    Returns:
+        [string]: Postgres connection string
+    """
+
+    return _create_postgres_url(
+        db_user=dbconfig.get('user'),
+        db_password=dbconfig.get('password'),
+        db_name=dbconfig.get('dbname'),
+        db_host=dbconfig.get('host'),
+        db_port=dbconfig.get('port'),
+        # change the SSL stuff once we have a real DB to connect to
+        db_ssl_mode='',
+        db_root_cert=''
+    )
+
+
+def create_postgres_url_from_env():
+    """Create postgres connection string from environment settings
+
+    Returns:
+        [string]: Postgres connection string
+    """
+
+    return _create_postgres_url(
+        db_user=get_environ_or_bust('POSTGRES_USER'),
+        db_password=get_environ_or_bust('POSTGRES_PASSWORD'),
+        db_name=get_environ_or_bust('POSTGRES_DB'),
+        db_host=get_environ_or_bust('POSTGRES_HOST'),
+        db_port=os.environ.get('POSTGRES_PORT', 5432),
+        db_ssl_mode=os.environ.get('POSTGRES_SSL_MODE', None),
+        db_root_cert=os.environ.get('POSTGRES_SSL_ROOT_CERTIFICATE', None)
+    )
+
+
+def init_db(dbconfig=None, outputfile=None, echo=None,
+            hide_parameters=None, create_all=False):
+    # import all modules here that might define models so that
+    # they will be registered properly on the metadata.  Otherwise
+    # you will have to import them first before calling init_db()
+    """Setup DB scoped session
+
+    Args:
+        config (dict): A dictionary of config settings
+            that are required to connect to the Postgres DB
+        outputfile (filepath): File location to log SQL statements
+        echo (bool): Whether or not all statements are logged to the
+            default log handler
+            https://docs.sqlalchemy.org/en/14/core/engines.html#sqlalchemy.create_engine.params.echo
+        hide_parameters (bool): if false then statement parameters
+            will not be logged to INFO leverl log messages or in
+            logged representation of error reports.
+            https://docs.sqlalchemy.org/en/14/core/engines.html#sqlalchemy.create_engine.params.hide_parameters
+
+    Returns:
+        [scoped_session]: DB scoped_session to use for DB SQL operations
+    """
+
+    # Setup a dedicated DB logger if a target output file was provided
+    create_db_logger(outputfile)
+
+    # Create the DB engine, either from provided settings, or
+    # using relevant environment settings.
+    if dbconfig:
+        engine_url = create_postgres_url_from_config(dbconfig)
+    elif os.environ.get('DATABASE_URI', None):
+        engine_url = os.environ['DATABASE_URI']
+    else:
+        engine_url = create_postgres_url_from_env()
+
+    # TODO(rtamalin): Remove this try/except hackery once we move forward
+    # to being based on SLE 15 SP3 or later.
```

Comment:
Per comment this can be cleaned up, we are not going to build/use this on anything older the 15 SP6